src/DifficultGame.py

```diff
@@ -0,0 +1,54 @@
+from WinningCombo import checkForWin
+from copy import deepcopy
+
+def bestComputerSpot (gameBoard):
+    computerPicks = [checkForGameEnder(deepcopy(gameBoard), "O"),
+                    checkForGameEnder(deepcopy(gameBoard), "X"),
+                    checkForSpotFive(deepcopy(gameBoard)),
+                    eliminateWinningCombo(deepcopy(gameBoard)),
+                    takeFirstAvailableSpot(deepcopy(gameBoard))]
+
+    for pick in computerPicks:
+        if pick:
+            return pick
+
+def checkForSpotFive(gameBoard):
+    if gameBoard[1][1] != "X" and gameBoard[1][1] !="O":
+        gameBoard[1][1] = "O"
+        return gameBoard
+
+def checkForGameEnder(gameboard, XorO):
```

Comment:
Would this violate DRY? Not sure what to make of this one. It is the same function but being used differently I guess. It would make it more readable but not sure which rule outweighs the other.

Reply:
Two helper functions were created, tryToWinGame() and blockWinningSpot(). These helper functions take in the gameboard and assign their corresponding move to checkForGameEnder(). The return for both is an augmented gameboard and have been placed in their respective bestComputerSpot() position.